### PR TITLE
Tiltfile: Support appended commands in Dockerfile

### PR DESCRIPTION
**Pull request by randomvariable**
The Docker provider image requires extra binaries in order to work, so add
provider fields to support appending commands to the Dockerfile contents.

Also adds buildifier as the Starlark formatter.

### CODE REVIEW

File: docs/book/src/developer/tilt.md [hunk 1]
```diff
@@ -108,6 +108,23 @@ build it.
...
 
+**additional_helper_image_commands** (String, default=""): Additional commands to be run in the helper image
+docker build. e.g.
+
+``` Dockerfile
+RUN wget -qO- https://dl.k8s.io/v1.14.4/kubernetes-client-linux-amd64.tar.gz | tar xvz
+RUN wget -qO- https://get.docker.com | sh
+```
+
+**additional_image_commands** (String, default=""): Additional commands to be appended to the dockerfile.
```
Comment (ncdc) [inline docs/book/src/developer/tilt.md]: Field name mismatch

File: docs/book/src/developer/tilt.md [hunk 2]
```diff
@@ -108,6 +108,23 @@ build it.
 **live_reload_deps**: a list of files/directories to watch. If any of them changes, Tilt rebuilds the manager binary
 for the provider and performs a live update of the running container.
 
+**additional_helper_image_commands** (String, default=""): Additional commands to be run in the helper image
```
Comment (ncdc) [inline docs/book/src/developer/tilt.md]: Field name mismatch

Comment (randomvariable) [reply docs/book/src/developer/tilt.md]: thanks!

**Comment by vincepri** [comment]
Shouldn't the providers maybe live under `tilt-provider.json` file instead? So other providers could use this feature as well

**Comment by randomvariable** [comment]
> Shouldn't the providers maybe live under tilt-provider.json file instead? So other providers could use this feature as well

That's how it works, hence the docs update, but the Docker provider is included inside the Tiltfile.

File: docs/book/src/developer/tilt.md [hunk 3]
```diff
@@ -108,6 +108,24 @@ build it.
...
+**additional_docker_build_commands** (String, default=""): Additional commands to be appended to
+the dockerfile.
+The manager image will use docker-slim, so to download files, use `additional_helper_image_commands`. e.g.
+
+``` Dockerfile
+COPY --from=tilt-helper /usr/bin/docker /usr/bin/docker
+COPY --from=tilt-helper /go/kubernetes/client/bin/kubectl /usr/bin/kubectl
+```
+
+
```
Comment (detiber) [inline docs/book/src/developer/tilt.md]:
```suggestion
```

**Comment by randomvariable** [comment]
/hold
Going to add the linter, which has changed its mind about `"` in the latest release :(

File: hack/verify-starlark.sh [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    cat "${OUT}"
+    ret=1
+  fi
+  echo "Cleaning up..."
+  rm -rf "${TMP_DIR}"
+  exit ${ret}
+}
+trap cleanup EXIT
+
+# install shellcheck (x64 only!)
```
Comment (ncdc) [inline hack/verify-starlark.sh]: copy pasta 😄

File: hack/verify-starlark.sh [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+cd "${TMP_DIR}" || exit
+
+curl -L "https://github.com/bazelbuild/buildtools/releases/download/${VERSION}/${BINARY}" -o buildifier
+chmod +x buildifier
+
+echo "Running buildifier..."
+cd "${ROOT_PATH}" || exit
+OUT="${TMP_DIR}/out.log"
+FILES=$(find . -name "*.bzl")
```
Comment (ncdc) [inline hack/verify-starlark.sh]: Do we need this .bzl glob?

**Comment by ncdc** [comment]
Can you also add a hack script + make target to run the formatter?

File: hack/verify-starlark.sh (hunk 5 above)
Comment (randomvariable) [reply hack/verify-starlark.sh]: Nope

File: hack/verify-starlark.sh (hunk 4 above)
Comment (randomvariable) [reply hack/verify-starlark.sh]: :)

**Comment by randomvariable** [comment]
Added make target for the formatter.

File: hack/verify-starlark.sh [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  echo "Cleaning up..."
+  rm -rf "${TMP_DIR}"
+  exit ${ret}
+}
+trap cleanup EXIT
+
+# install buildifier
+cd "${TMP_DIR}" || exit
+
+curl -L "https://github.com/bazelbuild/buildtools/releases/download/${VERSION}/${BINARY}" -o buildifier
```
Comment (ncdc) [inline hack/verify-starlark.sh]: Can we put this in one of the gitignored directories and only download it if it doesn't already exist? Could use hack/tools/bin perhaps?

Comment (randomvariable) [reply hack/verify-starlark.sh]: Done. Also for shellcheck.

File: hack/verify-starlark.sh [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+# create a temporary directory
+TMP_DIR=$(mktemp -d)
+OUT="${TMP_DIR}/out.log"
+
+# cleanup on exit
+cleanup() {
+  ret=0
+  if [[ -s "${OUT}" ]]; then
+    echo "Found errors:"
```
Comment (ncdc) [inline hack/verify-starlark.sh]: Sorry, 1 last suggestion: mention `run make format-tiltfile` to fix the formatting errors

**Comment by ncdc** [comment]
@vincepri PTAL

File: hack/verify-starlark.sh (hunk 7 above)
Comment (randomvariable) [reply hack/verify-starlark.sh]: Done

**Pull request merged**